src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs

```diff
@@ -363,6 +387,35 @@ private static Block RebuildCodeGenerators(Block block)
             return builder.Build();
         }
 
+        private static SourceLocation GetAttributeNameStartLocation(SyntaxTreeNode node)
+        {
+            var startOffset = node.Start;
+
+            while (node.IsBlock)
+            {
+                // We want to track the start of the parent node. Each iteration of this while loop we will dive deeper
+                // into the Block. Since the attribute name will always be located at .Children[0] we need to track the
+                // corresponding .Start location of the attribute names parent so we can offset its .Start based on the 
+                // leading whitespace.
+                startOffset = node.Start;
+
+                // An attribute must have at least one child element underneath it
+                node = ((Block)node).Children.First();
+            }
+
+            // At this point the node is not a block so it must be a Span.
+            var span = (Span)node;
```

Comment:
more succinct and perhaps more readable as:
``` c#
  var span = (node as Block)?.FindFirstDescendentSpan() ?? (Span)node;
  var blockStart = span.Parent.Start;
```
worst case, use an `if` to set `span`.